src/object.d

```diff
@@ -3190,11 +3199,24 @@ void __ctfeWriteln(T...)(auto ref T values) { __ctfeWrite(values, "\n"); }
  * Create RTInfo for type T
  */
 
+template RTInfoImpl(size_t[] pointers)
+{
+    immutable size_t[pointers.length] data = pointers[];
+    immutable RTInfoImpl = data.ptr;
```

Comment:
> Ah right, that means you get the length from the type itself or from the typeinfo?

The length is stored in the first word of the pointer bitmap.

> In the later case it would be another random memory access.
> If we already have the length, then of course we don't need to store it.

Getting the length from the TypeInfo needs a virtual function call. It's already expensive enough having to do this for RTInfo.

> The alignment for ubyte* is only a single byte, so we can really squash this. Section alignment is only done when merging sections of different object files, and even then --sort-section=alignment is a simple linker optimization.

That option isn't available on every platform. We could use a separate section with 1 byte alignment, though.

> Maybe we need to rework the weird BlkAttr.APPENDABLE implementation?
> If we can leverage the information then it makes sense to move most array functionality from rt.lifetime into the GC.

Yes, BlkAttr.APPENDABLE already leaks some detail of the array implementation. Moving appending into the GC seems a good plan.

> Let's assume large objects with GC references are rare for now.

Arrays (e.g. of strings) can often get quite large. But I'm ok with leaving this optimization for later.